File: java-semantickernel/src/main/java/com/codeyou/agent/TimePlugin.java
package com.codeyou.agent;

import com.microsoft.semantickernel.semanticfunctions.annotations.DefineKernelFunction;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

/**
 * Plugin for time-related functions
 */
public class TimePlugin {
    
    @DefineKernelFunction(
        name = "getCurrentTime",
        description = "Gets the current date and time",
        returnDescription = "The current date and time as a string"
    )
    public String getCurrentTime() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
        return LocalDateTime.now().format(formatter);
    }
}

File: dotnet-semantickernel/TimePlugin.cs
using System.ComponentModel;
using Microsoft.SemanticKernel;

namespace SemanticKernelAgent
{
  /// <summary>
  /// Plugin for time-related functions
  /// </summary>
  public class TimePlugin
    {
        [KernelFunction, Description("Gets the current date and time")]
        public string GetCurrentTime()
        {
            return DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
        }
    }
}
